transit_odp/timetables/dataframes.py

```diff
@@ -14,6 +14,94 @@
 logger = logging.getLogger(__name__)
 
 
+def flexible_journey_patterns_to_dataframe(services):
+    flexible_journey_patterns = []
+    for service in services:
+        service_code = service.get_element(["ServiceCode"]).text
+        flexible_services = service.get_elements_or_none(["FlexibleService"])
+        if flexible_services:
+            for flexible_service in flexible_services:
+                for pattern in flexible_service.get_elements(
+                    ["FlexibleJourneyPattern"]
+                ):
+                    flexible_zone = pattern.get_element_or_none("FlexibleZones")
+                    direction = pattern.get_element_or_none("Direction").text
+                    direction = direction if direction else ""
+                    if flexible_zone:
+                        for flexistopusage in flexible_zone.get_elements(
```

Comment:
Block of code within this can be refactored and reused for fixed flexible stops and stops in sequence

Reply:
done pls check